src/main/java/seedu/duke/commands/ViewCommand.java

```diff
@@ -36,22 +36,23 @@ public void execute(NormalFinanceManager normalFinanceManager,
 
             outputArray = normalFinanceManager.getCopyOfArray();
             outputArray = recurringFinanceManager.appendEntryForView(viewOptions, outputArray, recurringOutputArray);
-            view(outputArray, recurringOutputArray, ui, viewOptions.isViewAll);
+            view(outputArray, recurringOutputArray, ui, viewOptions.isViewAll, viewOptions.isViewExpenseOrIncome);
         } catch (MintException e) {
             ui.printError(e);
         }
     }
 
     public void view(ArrayList<Entry> outputArray, ArrayList<Entry> recurringOutputArray,
-                     Ui ui, boolean isViewAll) throws MintException {
+                     Ui ui, boolean isViewAll, boolean isViewExpenseOrIncome) throws MintException {
         outputArray.sort(Sorter.compareByDate);
         recurringOutputArray.sort(Sorter.compareByDate);
         applyModifiers(outputArray);
         applyRecurringModifiers(recurringOutputArray);
 
         double total = calculateTotal(outputArray);
         int[] indentations = ui.printView(outputArray, viewOptions.fromDate, viewOptions.endDate, total);
-        ui.printViewRecurring(recurringOutputArray, indentations[0], indentations[1], indentations[2], isViewAll);
+        ui.printViewRecurring(recurringOutputArray, indentations[0], indentations[1], indentations[2], isViewAll,
```

Comment:
Is it better to declare variables to show what are the contents in the indentations Array? Or perhaps adding a comment to remind the developer.

Reply:
As I only added the last one parameter and the indentations array are @yanjia1777 's expertise, perhaps @yanjia1777 can help to resolve this later.

Reply:
@irvinseet i have to return multiple values since the printView call is done in viewCommand but the calculation of the indentations is done in Ui. thats why I had it return an array of multiple indentation values

Reply:
@irvinseet I can assign each value of the array to a variable first if it makes it clearer, like int catIndentation = indentations[0]